content/post/gld.md

```diff
@@ -0,0 +1,276 @@
+
+## Merged licenses
+
+There are projects with several license texts in the same file
+Some of them are dual-licensed, some mention dependencies. Our core matcher assumes the single
+sample by design, and those cases are hard for it to handle.
+[google/licenseclassifier](https://github.com/google/licenseclassifier) project gracefully
+digests them because it is based on ngram hashing instead, and hence we considered switching to that
+algorithm. We did not for the following reasons:
+
+1. The memory consumption is higher, about [200 MB](https://github.com/google/licenseclassifier/blob/master/classifier.go#L188).
+2. Non-trivial database preprocessing and lack of high-level documentation.
+3. Slower performance on single licenses.
+3. It appeared that 95% of the cases could be resolved by simple split heuristics.
+
+After all, we found that since it is very cheap to query a text, we could make a few split assumptions
+and process each variant independently. For example, many texts separate licenses
+with `===` or `---` decorations. Besides, it is common to place the license name before the body
+and we easily find all possible split points. This is still not bulletproff as in license-classifier,
+but as was written works reasonably well.
+
+## Pointers
+
+Sometimes, all the efforts fail and we do not discover anything in `LICENSE` files and in `README` files.
+There is still hope: we discovered that many projects contain the URL to the official license text
+in them. E.g. [awesome lists which have the CC0 badge at the bottom](https://github.com/terryum/awesome-deep-learning-papers),
+[Apache banners](https://github.com/dmlc/xgboost/blob/master/LICENSE) or numerous users of
+[mit-license.org](https://mit-license.org). 
+
+## Implementation
+
+go-license-detector is a library and a self-contained binary CLI tool.
+
+[Francesc](https://twitter.com/francesc), our VP of Developer Community, took a serious effort
+in making the code idiomatic. You may have experienced this: when you are focused on **what** your
+code is doing, you often miss **how** the code is looking. I would like to write
+a follow-up post which describes which points were improved and what were the
+typical issues.
+
+## Offtopic
+
+Since we had to manually look through hundreds of most-starred projects on GitHub, we noticed
+a few funny trends. Many Chinese repositories isolated from the other communities,
+awesome list expansion and others. Again, I should devote a separate post to those,
+they are funny and also help to understand the picture of open source popularity better.
+
+## PGA license survey
+
+We've recently released [Public Git Archive](https://github.com/src-d/datasets/PublicGitArchive) (PGA),
+182,000 Git repositories belonging to most popular projects on GitHub. It's index file contains the licenses
+detected by go-license-detector. The following pie chart summarizes the license usage in PGA:
+
+<svg id="pga-licenses"></svg>
```

Comment:
We agreed on using [{% codepen %} snippet](https://blog.sourced.tech/documentation/shortcodes/#codepen) for runnable code, to be hosted on codepen, as done for [lapjv post](https://blog.sourced.tech/post/lapjv)
I think @marnovo can provide the credentials, but for testing you can try with your our account ;)

Reply:
David, the code is needed to draw the svg (d3). I have two options: export the svg and use it as an image or use codepen but without any borders, titles, etc. Is it possible with codepen?

Reply:
**imho**, id does not worth to use two scripts just to render a static image as we're doing now.
I see two points of view:
- (a) the script is useful and it could be used by the reader &rarr; we could share it &rarr; codepen is the better way to do it,
- (b) the solely purpose of the scripts is to render an image &rarr; then let's host it, and get rid of the scripts

If you ask me, I think that the `/post/gld/chart.js` script you developed can be reused, so it could be interesting for the reader, so I'd vote for (a).